pkg/daemon/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,355 @@
+		servicesCMExists          bool
+		configMapServices         []servicescm.Service
+	}{
+		// Tests with a valid services ConfigMap in the cluster (happy path)
+		{
+			name:                      "No services",
+			existingServices:          map[string]*fake.FakeService{},
+			expectedRemovedServices:   []string{},
+			expectedRemainingServices: []string{},
+			preserveNode:              false,
+			servicesCMExists:          true,
+			configMapServices:         []servicescm.Service{},
+		},
+		{
+			name:                      "Upgrade scenario with no services",
+			existingServices:          map[string]*fake.FakeService{},
+			expectedRemovedServices:   []string{},
+			expectedRemainingServices: []string{},
+			preserveNode:              true,
+			servicesCMExists:          true,
+			configMapServices:         []servicescm.Service{},
+		},
+		{
+			name:                      "Defined service doesn't exist on node",
+			existingServices:          map[string]*fake.FakeService{},
+			expectedRemovedServices:   []string{"test1"},
+			expectedRemainingServices: []string{},
+			preserveNode:              false,
+			servicesCMExists:          true,
+			configMapServices: []servicescm.Service{
+				{
+					Name:    "test1",
+					Command: "test1 --node-name=NODENAME",
+					NodeVariablesInCommand: []servicescm.NodeCmdArg{{
+						Name:               "NODENAME",
+						NodeObjectJsonPath: "{.metadata.name}",
+					}},
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     0,
+				},
+			},
+		},
+		{
+			name: "Single service",
+			existingServices: map[string]*fake.FakeService{
+				"test1": fake.NewFakeService(
+					"test1",
+					mgr.Config{
+						BinaryPathName: "test1 --node-name=node",
+						Dependencies:   []string{},
+						Description:    windows.ManagedTag + " test1",
+					},
+					svc.Status{State: svc.Running},
+				),
+			},
+			expectedRemovedServices:   []string{"test1"},
+			expectedRemainingServices: []string{},
+			preserveNode:              false,
+			servicesCMExists:          true,
+			configMapServices: []servicescm.Service{
+				{
+					Name:    "test1",
+					Command: "test1 --node-name=NODENAME",
+					NodeVariablesInCommand: []servicescm.NodeCmdArg{{
+						Name:               "NODENAME",
+						NodeObjectJsonPath: "{.metadata.name}",
+					}},
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     0,
+				},
+			},
+		},
+		{
+			name: "Upgrade scenario with multiple services",
+			existingServices: map[string]*fake.FakeService{
+				"test1": fake.NewFakeService(
+					"test1",
+					mgr.Config{
+						BinaryPathName: "test1 arg1",
+						Dependencies:   []string{},
+						Description:    windows.ManagedTag + " test1",
+					},
+					svc.Status{State: svc.Running},
+				),
+				"test2": fake.NewFakeService(
+					"test2",
+					mgr.Config{
+						BinaryPathName: "test2 arg1 arg2",
+						Dependencies:   []string{},
+						Description:    windows.ManagedTag + " test2",
```

Comment:
BinaryPathName doesn't really serve a purpose here right? It could be any value?
Might be helpful to add a helper function to reduce the bloat of these test cases, and making adding future cases easier
```
func newConfig(name string, managed bool, dependencies []string) mgr.Config {
  description: ""
  if managed {
    description = windows.Managed + " name"
  }
  return mgr.Config {
    BinaryPathName: "name",
    Description: description
    Dependencies: dependencies
  }
}
```

Reply:
Added a func to return a fake service altogether